Clinical-Genomics/scout
# Re-write ClinVar module to prepare for ClinVar API submissions

Fix #3569 and fix #3602

<img width="1115" alt="image" src="https://user-images.githubusercontent.com/28093618/193233293-5e8731d9-b2bb-4701-b317-5ca44de948d5.png">


- Doesn't change the functionality, but makes it **more user-friendly**
- Allow for submissions with phenotype type from the following DBs: [OMIM, MedGen, Orphanet, MeSH, HP, MONDO]
- The only change is that it is possible to add one variant at the time to ClinVar submissions
- Simplifies the fields exported in CaseData .CSV files (most of the current stuff is not used anyway)
- Moves all ClinVar export code to a dedicated "clinvar" blueprint. Before is was half under "variant" and half under "institutes"
- Is Using Flask-form to define the ClinVar input fields to be saved for a Variant submission, instead of ugly HTML code
- Created a model under scout.models that describes exactly what the expected formats are when you create and save a submission in the database
- Increases the test coverage

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>


**How to test**:
1. Install on cg-vm1 and try to save variants to ClinVar submissions and edit/delete ClinVar submissions for an institute
2. Try to download the Variants and CaseData .csv files from the Institute's submissions page and make sure download works

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
